src/csharp/Microsoft.Spark/Interop/Ipc/SerDe.cs

```diff
@@ -322,6 +322,13 @@ public static void Write(Stream s, long value)
         public static void Write(Stream s, double value) =>
             Write(s, BitConverter.DoubleToInt64Bits(value));
 
+        /// <summary>
+        /// Writes a decimal to a stream as a string.
+        /// </summary>
+        /// <param name="s">The stream to write</param>
+        /// <param name="value">The decimal to write</param>
+        public static void Write(Stream s, decimal value) => Write(s, value.ToString());
```

Comment:
Should use ToString(CultureInfo.InvariantCulture) if we are using a string on the wire.